### PR TITLE
run tests in travis

### PR DESCRIPTION
**Pull request by ankilosaurus**
Making tests run in travis build. (kubeconfig is not available on travis node. So fake clientset)

### CODE REVIEW

**Comment by grosser** [comment]
where are all these new dependencies coming from ?

**Comment by ankilosaurus** [comment]
> where are all these new dependencies coming from ?

fake import i believe

File: pkg/remediator/crashloopbackoff_rescheduler_test.go [hunk 1]
```diff
@@ -122,17 +124,21 @@ func (suite *TestCrashLoopBackOffReschedulerSuite) testRemediator() {
...
 	cancel() // cancel first so we can just run once and exit
 
+	suite.mockClient.EXPECT().GetSharedInformerFactory(suite.logger, "").Return(suite.testGetInformerFactory(), nil).Times(1)
 	crashLoopRemediator, err := suite.testGetRemediator()
 	assert.Equal(suite.t, err, nil)
 	assert.Assert(suite.t, crashLoopRemediator != nil)
 	crashLoopRemediator.Run(ctx, nil)
 }
 
+func (suite *TestCrashLoopBackOffReschedulerSuite) testGetInformerFactory() informers.SharedInformerFactory {
```
Comment (grosser) [inline pkg/remediator/crashloopbackoff_rescheduler_test.go:134]: not a test -> no test prefix ?
```suggestion
func (suite *TestCrashLoopBackOffReschedulerSuite) getInformerFactory() informers.SharedInformerFactory {
```

**Review by grosser — APPROVED** [review]

File: pkg/k8s/client.go [hunk 2]
```diff
@@ -30,6 +32,11 @@ func (c *Client) DeletePod(pod *apiv1.Pod) error {
 	return c.clientSet.CoreV1().Pods(pod.ObjectMeta.Namespace).Delete(pod.ObjectMeta.Name, &metav1.DeleteOptions{})
 }
 
+func (c *Client) GetSharedInformerFactory(logger *zap.Logger, ns string) (informers.SharedInformerFactory, error) {
```
Comment (grosser) [inline pkg/k8s/client.go]: logger argument is not used ?

File: pkg/remediator/crashloopbackoff_rescheduler.go [hunk 3]
```diff
@@ -142,7 +142,7 @@ func NewCrashLoopBackOffRescheduler(logger *zap.Logger,
 	metrics := metrics.NewCrashLoopBackOffMetrics(logger)
 	metrics.Register()
 
-	informerFactory, err := k8s.NewSharedInformerFactory(logger, filter.namespace)
+	informerFactory, err := client.GetSharedInformerFactory(logger, filter.namespace)
```
Comment (grosser) [inline pkg/remediator/crashloopbackoff_rescheduler.go]: it returns a new one each time ... so "New" not "Get"
```suggestion
	informerFactory, err := client.NewSharedInformerFactory(logger, filter.namespace)
```

File: pkg/http/http_test.go [hunk 4]
```diff
@@ -57,11 +58,13 @@ func (suite *TestHttpServerSuite) TestServer() {
 	wg.Add(1)
 	go remediator_http.NewServer(suite.logger).Serve(ctx, &wg)
 
+	time.Sleep(10 * time.Second)
```
Comment (grosser) [inline pkg/http/http_test.go]: prefer 0.1 or so, it should be fast

File: pkg/http/http_test.go [hunk 5]
```diff
@@ -57,11 +58,13 @@ func (suite *TestHttpServerSuite) TestServer() {
...
 	status, _ := suite.httpGet("http://localhost:8080/healthz")
 	assert.Equal(suite.t, status, 200)
 	status, _ = suite.httpGet("http://localhost:8080/metrics")
 	assert.Equal(suite.t, status, 200)
 
+	suite.logger.Sugar().Infof("closing")
```
Comment (grosser) [inline pkg/http/http_test.go]: can use fmt.PrintLn to debug

File: pkg/remediator/crashloopbackoff_rescheduler_test.go (hunk 1 above)
Comment (grosser) [reply pkg/remediator/crashloopbackoff_rescheduler_test.go:134]: still needs a rename ? `newInformerFactory `

File: pkg/http/http_test.go (hunk 5 above)
Comment (ankilosaurus) [reply pkg/http/http_test.go]: was just for debugging. will remove

File: pkg/http/http_test.go (hunk 4 above)
Comment (ankilosaurus) [reply pkg/http/http_test.go]: changed to 1sec

Comment (grosser) [reply pkg/http/http_test.go]: prefer even lower if it works ... keeping tests fast is nice :)

File: pkg/http/http_test.go [hunk 6]
```diff
@@ -57,6 +58,7 @@ func (suite *TestHttpServerSuite) TestServer() {
 	wg.Add(1)
 	go remediator_http.NewServer(suite.logger).Serve(ctx, &wg)
 
+	time.Sleep(1 * time.Second)
```
Comment (grosser) [inline pkg/http/http_test.go:62]: say why we are waiting

**Pull request merged**